projboard/models/user.py

```diff
@@ -28,3 +28,66 @@ def get_user_by_nickname(nickname):
         except ObjectDoesNotExist:
             return None
         return user
+
+    @staticmethod
+    def get_user(email, password, name, nickname):
+        """
+         Method to create a new user,
+         but also checks if user is already exists
+         [get user either by mail or by nickname]
+        :param email: user's email
+        :param password: user's password
+        :param name: user's name
+        :param nickname: user's nickname
+        :return: new user
+        """
+        try:
+            user = User.objects.get(nickname=nickname)
+            # user = User.objects.get(email=email)
+            return user
+        except User.DoesNotExist:
```

Comment:
You would never get this exception, because it is not raised anywhere.  
`get_user_by_nickname` is not rising anything, it catches the `ObjectDoesNotExist` and returns `None`

You should either change `get_user_by_nickname` to raise the exception or change your implementation